app/models/spree/adjustment.rb

```diff
@@ -106,8 +106,8 @@ def set_eligibility
     # more than on line items at once via accepted_nested_attributes the order
     # object on the association would be in a old state and therefore the
     # adjustment calculations would not performed on proper values
-    def update!(calculable = nil)
-      return if immutable?
+    def update!(calculable = nil, force: false)
+      return if immutable? && !force
```

Comment:
Usually the boolean flag pattern is a little code smelly; I think this particular case isn't too bad though.

Reply:
There's various places where we have all kinds of mess in the code that are basically workarounds for the fact that adjustments normally are not updatable when they're closed (after the order is complete). We actually break this rule often enough that it seems like having a simple and clear way to update closed adjustments would be more sensible than multiple hacks to get around it. We can get rid of the following code, and I think it's a good trade:

https://github.com/openfoodfoundation/openfoodnetwork/blob/7ad165ee3c287af5629e856f4ad1122e60795c79/app/models/spree/order.rb#L869-L873

https://github.com/openfoodfoundation/openfoodnetwork/blob/7ad165ee3c287af5629e856f4ad1122e60795c79/app/controllers/spree/orders_controller.rb#L159-L171